openai_embeddings does not support asymmetric embedding models for NVIDIA NIMs

### System Info

Python 3.12.11

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

When running the tests for openai_embeddings with ```nvidia/llama-3.2-nv-embedqa-1b-v2``` with ```https://integrate.api.nvidia.com``` base URL, I got the following error:

```shell
openai.BadRequestError: Error code: 400 - {'error': "Your request cannot be validated, it is incorrect. detail='input_type' parameter is required for asymmetric models"}
``` 

This is because the openai_embeddings method does not have the `input_type` parameter, but the asymmetric models need the `input_type` parameter for embeddings.

### Error logs

Here is the error log from one of the test cases in ```test_openai_embeddings```: 

```
_ test_openai_embeddings_with_user_parameter[llama_stack_client-emb=nvidia/llama-3.2-nv-embedqa-1b-v2] _
tests/integration/inference/test_openai_embeddings.py:155: in test_openai_embeddings_with_user_parameter
    response = compat_client.embeddings.create(
.venv/lib/python3.12/site-packages/llama_stack_client/resources/embeddings.py:89: in create
    return self._post(
.venv/lib/python3.12/site-packages/llama_stack_client/_base_client.py:1232: in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/distribution/library_client.py:198: in request
    result = loop.run_until_complete(self.async_client.request(*args, **kwargs))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../miniconda3/envs/llama/lib/python3.12/asyncio/base_events.py:691: in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
llama_stack/distribution/library_client.py:312: in request
    response = await self._call_non_streaming(
llama_stack/distribution/library_client.py:364: in _call_non_streaming
    result = await matched_func(**body)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/providers/utils/telemetry/trace_protocol.py:101: in async_wrapper
    result = await method(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/distribution/routers/inference.py:580: in openai_embeddings
    return await provider.openai_embeddings(**params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/providers/utils/inference/openai_mixin.py:226: in openai_embeddings
    response = await self.client.embeddings.create(
.venv/lib/python3.12/site-packages/openai/resources/embeddings.py:243: in create
    return await self._post(
.venv/lib/python3.12/site-packages/openai/_base_client.py:1767: in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/openai/_base_client.py:1461: in request
    return await self._request(
.venv/lib/python3.12/site-packages/openai/_base_client.py:1562: in _request
    raise self._make_status_error_from_response(err.response) from None
E   openai.BadRequestError: Error code: 400 - {'error': "Your request cannot be validated, it is incorrect. detail='input_type' parameter is required for asymmetric models"}
_ test_openai_embeddings_different_inputs_different_outputs[llama_stack_client-emb=nvidia/llama-3.2-nv-embedqa-1b-v2] _
tests/integration/inference/test_openai_embeddings.py:196: in test_openai_embeddings_different_inputs_different_outputs
    response1 = compat_client.embeddings.create(
.venv/lib/python3.12/site-packages/llama_stack_client/resources/embeddings.py:89: in create
    return self._post(
.venv/lib/python3.12/site-packages/llama_stack_client/_base_client.py:1232: in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/distribution/library_client.py:198: in request
    result = loop.run_until_complete(self.async_client.request(*args, **kwargs))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../miniconda3/envs/llama/lib/python3.12/asyncio/base_events.py:691: in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
llama_stack/distribution/library_client.py:312: in request
    response = await self._call_non_streaming(
llama_stack/distribution/library_client.py:364: in _call_non_streaming
    result = await matched_func(**body)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/providers/utils/telemetry/trace_protocol.py:101: in async_wrapper
    result = await method(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/distribution/routers/inference.py:580: in openai_embeddings
    return await provider.openai_embeddings(**params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/providers/utils/inference/openai_mixin.py:226: in openai_embeddings
    response = await self.client.embeddings.create(
.venv/lib/python3.12/site-packages/openai/resources/embeddings.py:243: in create
    return await self._post(
.venv/lib/python3.12/site-packages/openai/_base_client.py:1767: in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/openai/_base_client.py:1461: in request
    return await self._request(
.venv/lib/python3.12/site-packages/openai/_base_client.py:1562: in _request
    raise self._make_status_error_from_response(err.response) from None
E   openai.BadRequestError: Error code: 400 - {'error': "Your request cannot be validated, it is incorrect. detail='input_type' parameter is required for asymmetric models"}
_ test_openai_embeddings_base64_batch_processing[llama_stack_client-emb=nvidia/llama-3.2-nv-embedqa-1b-v2] _
tests/integration/inference/test_openai_embeddings.py:253: in test_openai_embeddings_base64_batch_processing
    response = compat_client.embeddings.create(
.venv/lib/python3.12/site-packages/llama_stack_client/resources/embeddings.py:89: in create
    return self._post(
.venv/lib/python3.12/site-packages/llama_stack_client/_base_client.py:1232: in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/distribution/library_client.py:198: in request
    result = loop.run_until_complete(self.async_client.request(*args, **kwargs))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../miniconda3/envs/llama/lib/python3.12/asyncio/base_events.py:691: in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
llama_stack/distribution/library_client.py:312: in request
    response = await self._call_non_streaming(
llama_stack/distribution/library_client.py:364: in _call_non_streaming
    result = await matched_func(**body)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/providers/utils/telemetry/trace_protocol.py:101: in async_wrapper
    result = await method(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/distribution/routers/inference.py:580: in openai_embeddings
    return await provider.openai_embeddings(**params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
llama_stack/providers/utils/inference/openai_mixin.py:226: in openai_embeddings
    response = await self.client.embeddings.create(
.venv/lib/python3.12/site-packages/openai/resources/embeddings.py:243: in create
    return await self._post(
.venv/lib/python3.12/site-packages/openai/_base_client.py:1767: in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/openai/_base_client.py:1461: in request
    return await self._request(
.venv/lib/python3.12/site-packages/openai/_base_client.py:1562: in _request
    raise self._make_status_error_from_response(err.response) from None
E   openai.BadRequestError: Error code: 400 - {'error': "Your request cannot be validated, it is incorrect. detail='input_type' parameter is required for asymmetric models"}
```

### Expected behavior

Expect to either have `openai_embeddings` method support the `input_type` parameter, or document that the method does not support asymmetric embedding method.

--- a/llama_stack/providers/remote/inference/nvidia/NVIDIA.md
+++ b/llama_stack/providers/remote/inference/nvidia/NVIDIA.md
@@ -77,6 +77,10 @@ print(f"Response: {response.completion_message.content}")
 ```
 
 ### Create Embeddings
+> Note on OpenAI embeddings compatibility
+>
+> NVIDIA asymmetric embedding models (e.g., `nvidia/llama-3.2-nv-embedqa-1b-v2`) require an `input_type` parameter not present in the standard OpenAI embeddings API. The NVIDIA Inference Adapter automatically sets `input_type="query"` when using the OpenAI-compatible embeddings endpoint for NVIDIA. For passage embeddings, use the `embeddings` API with `task_type="document"`.
+
 ```python
 response = client.inference.embeddings(
     model_id="nvidia/llama-3.2-nv-embedqa-1b-v2",

--- a/llama_stack/providers/remote/inference/nvidia/nvidia.py
+++ b/llama_stack/providers/remote/inference/nvidia/nvidia.py
@@ -8,7 +8,7 @@ import logging
 import warnings
 from collections.abc import AsyncIterator
 
-from openai import APIConnectionError, BadRequestError
+from openai import NOT_GIVEN, APIConnectionError, BadRequestError
 
 from llama_stack.apis.common.content_types import (
     InterleavedContent,
@@ -27,6 +27,9 @@ from llama_stack.apis.inference import (
     Inference,
     LogProbConfig,
     Message,
+    OpenAIEmbeddingData,
+    OpenAIEmbeddingsResponse,
+    OpenAIEmbeddingUsage,
     ResponseFormat,
     SamplingParams,
     TextTruncation,
@@ -209,6 +212,57 @@ class NVIDIAInferenceAdapter(OpenAIMixin, Inference, ModelRegistryHelper):
         # Llama Stack: EmbeddingsResponse(embeddings=list[list[float]])
         #
         return EmbeddingsResponse(embeddings=[embedding.embedding for embedding in response.data])
+
+    async def openai_embeddings(
+        self,
+        model: str,
+        input: str | list[str],
+        encoding_format: str | None = "float",
+        dimensions: int | None = None,
+        user: str | None = None,
+    ) -> OpenAIEmbeddingsResponse:
+        """
+        OpenAI-compatible embeddings for NVIDIA NIM.
+
+        Note: NVIDIA NIM asymmetric embedding models require an "input_type" field not present in the standard OpenAI embeddings API.
+        We default this to "query" to ensure requests succeed when using the
+        OpenAI-compatible endpoint. For passage embeddings, use the embeddings API with
+        `task_type='document'`.
+        """
+        extra_body: dict[str, object] = {"input_type": "query"}
+        logger.warning(
+            "NVIDIA OpenAI-compatible embeddings: defaulting to input_type='query'. "
+            "For passage embeddings, use the embeddings API with task_type='document'."
+        )
+
+        response = await self.client.embeddings.create(
+            model=await self._get_provider_model_id(model),
+            input=input,
+            encoding_format=encoding_format if encoding_format is not None else NOT_GIVEN,
+            dimensions=dimensions if dimensions is not None else NOT_GIVEN,
+            user=user if user is not None else NOT_GIVEN,
+            extra_body=extra_body,
+        )
+
+        data = []
+        for i, embedding_data in enumerate(response.data):
+            data.append(
+                OpenAIEmbeddingData(
+                    embedding=embedding_data.embedding,
+                    index=i,
+                )
+            )
+
+        usage = OpenAIEmbeddingUsage(
+            prompt_tokens=response.usage.prompt_tokens,
+            total_tokens=response.usage.total_tokens,
+        )
+
+        return OpenAIEmbeddingsResponse(
+            data=data,
+            model=response.model,
+            usage=usage,
+        )
 
     async def chat_completion(
         self,

